Segfault during `run --wait` when reading manifest from stdin/file

**What steps did you take and what happened:**
A segfault occurs when the output from `sonobuoy gen` is passed via file or stdin to `sonobuoy run`.

```
./sonobuoy gen --mode quick | ./sonobuoy run --wait -f -                                                                                                                                                                              0|1 ↵  7513  15:19:59
INFO[0000] created object                                name=sonobuoy namespace= resource=namespaces
INFO[0000] created object                                name=sonobuoy-serviceaccount namespace=sonobuoy resource=serviceaccounts
INFO[0000] created object                                name=sonobuoy-serviceaccount-sonobuoy namespace= resource=clusterrolebindings
INFO[0000] created object                                name=sonobuoy-serviceaccount namespace= resource=clusterroles
INFO[0000] created object                                name=sonobuoy-config-cm namespace=sonobuoy resource=configmaps
INFO[0000] created object                                name=sonobuoy-plugins-cm namespace=sonobuoy resource=configmaps
INFO[0000] created object                                name=sonobuoy namespace=sonobuoy resource=pods
INFO[0000] created object                                name=sonobuoy-master namespace=sonobuoy resource=services
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x180 pc=0x10a7e5e]

goroutine 1 [running]:
github.com/vmware-tanzu/sonobuoy/pkg/client.(*SonobuoyClient).RunManifest.func1(0xc0007e9a18, 0xc0007e99c8, 0x2)
	/go/src/github.com/vmware-tanzu/sonobuoy/pkg/client/run.go:104 +0x3e
github.com/vmware-tanzu/sonobuoy/vendor/k8s.io/apimachinery/pkg/util/wait.WaitFor(0xc0003dca40, 0xc0007e9c48, 0xc000628420, 0x0, 0x0)
	/go/src/github.com/vmware-tanzu/sonobuoy/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:405 +0x155
github.com/vmware-tanzu/sonobuoy/vendor/k8s.io/apimachinery/pkg/util/wait.pollInternal(0xc0003dca40, 0xc0007e9c48, 0x0, 0x0)
	/go/src/github.com/vmware-tanzu/sonobuoy/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:293 +0x9f
github.com/vmware-tanzu/sonobuoy/vendor/k8s.io/apimachinery/pkg/util/wait.Poll(0x4a817c800, 0x4e94914f0000, 0xc0007e9c48, 0x7, 0xffffffffffffffff)
	/go/src/github.com/vmware-tanzu/sonobuoy/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:287 +0x4d
github.com/vmware-tanzu/sonobuoy/pkg/client.(*SonobuoyClient).RunManifest(0xc000029e00, 0xc000456180, 0xc000314000, 0x11c2, 0x1e00, 0x0, 0x0)
	/go/src/github.com/vmware-tanzu/sonobuoy/pkg/client/run.go:130 +0x791
github.com/vmware-tanzu/sonobuoy/pkg/client.(*SonobuoyClient).Run(0xc000029e00, 0xc000456180, 0x23433e0, 0x0)
	/go/src/github.com/vmware-tanzu/sonobuoy/pkg/client/run.go:163 +0x99
github.com/vmware-tanzu/sonobuoy/cmd/sonobuoy/app.submitSonobuoyRun(0xc000139b80, 0xc00002e900, 0x0, 0x3)
	/go/src/github.com/vmware-tanzu/sonobuoy/cmd/sonobuoy/app/run.go:109 +0x1d5
github.com/vmware-tanzu/sonobuoy/vendor/github.com/spf13/cobra.(*Command).execute(0xc000139b80, 0xc00002e870, 0x3, 0x3, 0xc000139b80, 0xc00002e870)
	/go/src/github.com/vmware-tanzu/sonobuoy/vendor/github.com/spf13/cobra/command.go:830 +0x2aa
github.com/vmware-tanzu/sonobuoy/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0000de000, 0x0, 0x1364c80, 0xc0000ae058)
	/go/src/github.com/vmware-tanzu/sonobuoy/vendor/github.com/spf13/cobra/command.go:914 +0x2fb
github.com/vmware-tanzu/sonobuoy/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/github.com/vmware-tanzu/sonobuoy/vendor/github.com/spf13/cobra/command.go:864
main.main()
	/go/src/github.com/vmware-tanzu/sonobuoy/main.go:26 +0x27

```

This is happening because the wait logic tries to get the status of the aggregator pod using the [`config.Config.Namespace`](https://github.com/vmware-tanzu/sonobuoy/blob/master/pkg/client/run.go#L104). This config is only set in the case where we aren't using a file: https://github.com/vmware-tanzu/sonobuoy/blob/master/cmd/sonobuoy/app/run.go#L68.

**What did you expect to happen:**
For the run to complete successfully with no segfault.

**Environment:**
- Sonobuoy version: Sonobuoy Version: v0.16.1-2-g3290552

--- a/Makefile
+++ b/Makefile
@@ -90,6 +90,7 @@ stress: sonobuoy
 
 # Integration tests
 int: DOCKER_FLAGS=-v $(KUBECONFIG):/root/.kube/kubeconfig --env KUBECONFIG=/root/.kube/kubeconfig --network host
+int: TESTARGS= $(VERBOSE_FLAG) -timeout 3m
 int: sonobuoy
 	$(DOCKER_BUILD) 'CGO_ENABLED=0 $(INT_TEST)'
 

--- a/pkg/client/run.go
+++ b/pkg/client/run.go
@@ -57,6 +57,7 @@ func (c *SonobuoyClient) RunManifest(cfg *RunConfig, manifest []byte) error {
 	buf := bytes.NewBuffer(manifest)
 	d := yaml.NewYAMLOrJSONDecoder(buf, bufferSize)
 
+	var createdNamespace string
 	for {
 		ext := runtime.RawExtension{}
 		if err := d.Decode(&ext); err != nil {
@@ -84,6 +85,14 @@ func (c *SonobuoyClient) RunManifest(cfg *RunConfig, manifest []byte) error {
 		if err != nil {
 			return errors.Wrap(err, "could not get object namespace")
 		}
+
+		// The namespace value is required if Wait is enabled in order to check the status.
+		// It may not be available within the RunConfig but we can retrieve it from the
+		// namespace of objects within the manifest.
+		if createdNamespace == "" && namespace != "" {
+			createdNamespace = namespace
+		}
+
 		// err is used to determine output for user; but first extract resource
 		_, err = c.dynamicClient.CreateObject(obj)
 		resource, err2 := c.dynamicClient.ResourceVersion(obj)
@@ -101,7 +110,7 @@ func (c *SonobuoyClient) RunManifest(cfg *RunConfig, manifest []byte) error {
 		seenStatus := false
 		runCondition := func() (bool, error) {
 			// Get the Aggregator pod and check if its status is completed or terminated.
-			status, err := c.GetStatus(&StatusConfig{cfg.Config.Namespace})
+			status, err := c.GetStatus(&StatusConfig{Namespace: createdNamespace})
 			switch {
 			case err != nil && seenStatus:
 				return false, errors.Wrap(err, "failed to get status")

--- a/test/integration/sonobuoy_integration_test.go
+++ b/test/integration/sonobuoy_integration_test.go
@@ -6,6 +6,7 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	"io/ioutil"
 	"os"
 	"os/exec"
 	"strings"
@@ -68,6 +69,45 @@ func cleanup(t *testing.T, namespace string) {
 		t.Logf("Error encountered during cleanup: %q\n", err)
 		t.Log(stdout.String())
 		t.Log(stderr.String())
+	}
+}
+
+func TestUseNamespaceFromManifest(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+
+	ns := "sonobuoy-" + strings.ToLower(t.Name())
+	defer cleanup(t, ns)
+
+	genArgs := fmt.Sprintf("gen -p testImage/yaml/job-junit-passing-singlefile.yaml -n %v", ns)
+	err, genStdout, genStderr := runSonobuoyCommandWithContext(ctx, t, genArgs)
+
+	if err != nil {
+		t.Fatalf("Sonobuoy exited with an error: %q\n", err)
+		t.Log(genStderr.String())
+	}
+
+	// Write the contents of gen to a temp file
+	tmpfile, err := ioutil.TempFile("", "gen.*.yaml")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer os.Remove(tmpfile.Name()) // clean up
+
+	if _, err := tmpfile.Write(genStdout.Bytes()); err != nil {
+		t.Fatal(err)
+	}
+	if err := tmpfile.Close(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Pass the gen output to sonobuoy run
+	runArgs := fmt.Sprintf("run --wait -f %v", tmpfile.Name())
+	err, _, runStderr := runSonobuoyCommandWithContext(ctx, t, runArgs)
+	if err != nil {
+		t.Errorf("Sonobuoy exited with an error: %q\n", err)
+		t.Log(runStderr.String())
 	}
 }
 

